testsuite/classlibrary/TestInteger.sc

```diff
@@ -0,0 +1,25 @@
+TestInteger : UnitTest {
+	test_forBy_positiveIntegerArguments {
+		var forByArguments = List.new;
+		10.forBy(20, 3, { | i, j | forByArguments.add([i, j]); });
+		this.assertEquals(forByArguments, List[[10, 0], [13, 1], [16, 2], [19, 3]]);
+	}
+
+	test_forBy_negativeIntegerArguments {
+		var forByArguments = List.new;
+		forBy(3, -7, -2, { | i, j | forByArguments.add([i, j]); });
+		this.assertEquals(forByArguments, List[[3, 0], [1, 1], [-1, 2], [-3, 3], [-5, 4], [-7, 5]]);
+	}
+
+	test_forBy_positiveFloatArguments {
+		var forByArguments = List.new;
+		forBy(-6, -4, 0.5, { | i, j | forByArguments.add([i, j]); });
+		this.assertEquals(forByArguments, List[[-6.0, 0], [-5.5, 1], [-5.0, 2], [-4.5, 3], [-4.0, 4]]);
+	}
+
+	test_forBy_negativeFloatArguments {
+		var forByArguments = List.new;
+		21.forBy(14.5, -3.125, { | i, j | forByArguments.add([i, j]); });
+		this.assertEquals(forByArguments, List[[21.0, 0], [17.875, 1], [14.75, 2]]);
+	}
+}
```

Comment:
An unfortunate bug that hasn't been addressed - the IDE doesn't append a newline to the end of the final line of a document, so most files edited with it will appear to be missing a final newline in git. I looked into it a couple months ago but didn't really come up with anything, and ended up moving on to other issues. Something to keep in mind for the future